src/server/pegasus_server_impl.cpp

```diff
@@ -3007,6 +3013,96 @@ void pegasus_server_impl::reset_usage_scenario_options(
     target_opts->max_write_buffer_number = base_opts.max_write_buffer_number;
 }
 
+void pegasus_server_impl::recalculate_data_cf_options(
+    const rocksdb::ColumnFamilyOptions &cur_data_cf_opts)
+{
+#define UPDATE_NUMBER_OPTION_IF_NEEDED(option, value)                                              \
+    do {                                                                                           \
+        auto _v = (value);                                                                         \
+        if (_v != cur_data_cf_opts.option) {                                                       \
+            new_options[#option] = std::to_string(_v);                                             \
+        }                                                                                          \
+    } while (0)
+
+#define UPDATE_BOOL_OPTION_IF_NEEDED(option, value)                                                \
+    do {                                                                                           \
+        if ((value) != cur_data_cf_opts.option) {                                                  \
```

Comment:
Should `(value)` be assigned to a variable (e.g. `auto _v = (value);`), just like what has been done in `UPDATE_NUMBER_OPTION_IF_NEEDED`, to avoid evaluating 2 times ?